x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```
@@ -0,0 +1,459 @@
+                                            public boolean advanceExact(int doc) throws IOException {
+                                                return values.advanceExact(doc);
+                                            }
+
+                                            @Override
+                                            public HistogramValue histogram() {
+                                                try {
+                                                    return getHistogramValue(values.binaryValue());
+                                                } catch (IOException e) {
+                                                    throw new IllegalStateException("Cannot load doc value", e);
+                                                }
+                                            }
+                                        };
+                                    } catch (IOException e) {
+                                        throw new IllegalStateException("Cannot load doc values", e);
+                                    }
+
+                                }
+
+                                @Override
+                                public ScriptDocValues<?> getScriptValues() {
+                                    return new ScriptDocValues.Strings(getBytesValues());
+                                }
+
+                                @Override
+                                public SortedBinaryDocValues getBytesValues() {
+                                    try {
+                                        final BinaryDocValues values = DocValues.getBinary(context.reader(), fieldName);
+                                        return FieldData.singleton(values);
+                                    } catch (IOException e) {
+                                        throw new IllegalStateException("Cannot load doc values", e);
+                                    }
+                                }
```

We should throw an UnsupportedOperationException in the two above methods.